quickwit-doc-mapper/src/tokenizers.rs

```diff
@@ -18,13 +18,106 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 use once_cell::sync::Lazy;
-use tantivy::tokenizer::{RawTokenizer, RemoveLongFilter, TextAnalyzer, TokenizerManager};
+use regex::Regex;
+use std::str::CharIndices;
+use tantivy::tokenizer::{
+    BoxTokenStream, RawTokenizer, RemoveLongFilter, TextAnalyzer, Token, TokenStream, Tokenizer,
+    TokenizerManager,
+};
+
+/// Tokenize the text without splitting on ".", "-" and "_" in numbers.
+#[derive(Clone)]
+pub struct LogTokenizer;
+
+pub struct LogTokenStream<'a> {
+    text: &'a str,
+    chars: CharIndices<'a>,
+    token: Token,
+}
+
+impl Tokenizer for LogTokenizer {
+    fn token_stream<'a>(&self, text: &'a str) -> BoxTokenStream<'a> {
+        BoxTokenStream::from(LogTokenStream {
+            text,
+            chars: text.char_indices(),
+            token: Token::default(),
+        })
+    }
+}
+
+impl<'a> LogTokenStream<'a> {
+    fn search_token_end(&mut self) -> usize {
+        (&mut self.chars)
+            .filter(|&(_, ref c)| !c.is_alphanumeric())
+            .map(|(offset, _)| offset)
+            .next()
+            .unwrap_or_else(|| self.text.len())
+    }
+
+    fn search_number_end(&mut self) -> usize {
+        let valid_chars_in_number = Regex::new("[-_.:a-zA-Z]").unwrap();
```

Comment:
You should not do a `Regex::new` every time the function is called. To avoid that, you can use `use once_cell::sync::Lazy` and create a static variable `static MY_REGEX: Lazy<Regex> =
        Lazy::new(|| Regex::new(MY_REGEX_CHARS).unwrap());`